pkg/manager/impl/execution_manager.go

```diff
@@ -263,6 +265,41 @@ func assignResourcesIfUnset(ctx context.Context, identifier *core.Identifier,
 	return resourceEntries
 }
 
+func checkTaskRequestsLessThanLimits(ctx context.Context, identifier *core.Identifier,
+	taskResources *core.Resources) {
+	// We choose the minimum of the platform request defaults or the limit itself for every resource request.
+	// Otherwise we can find ourselves in confusing scenarios where the injected platform request defaults exceed a
+	// user-specified limit
+	resourceLimits := make(map[core.Resources_ResourceName]string)
+	for _, resourceEntry := range taskResources.Limits {
+		resourceLimits[resourceEntry.Name] = resourceEntry.Value
+	}
+
+	finalizedResourceRequests := make([]*core.Resources_ResourceEntry, 0, len(taskResources.Requests))
+	for _, resourceEntry := range taskResources.Requests {
+		value := resourceEntry.Value
+		quantity := resource.MustParse(resourceEntry.Value)
+		limitValue, ok := resourceLimits[resourceEntry.Name]
+		if !ok {
+			// Unexpected - at this stage both requests and limits should be populated.
+			logger.Warningf(ctx, "No limit specified for [%v] resource [%s] although request was set", identifier,
+				resourceEntry.Name)
+			continue
+		}
+		if quantity.Cmp(resource.MustParse(limitValue)) == 1 {
+			// The quantity is greater than the limit! Course correct below.
+			logger.Debugf(ctx, "Updating requested value for task [%+v] resource [%s]. Overriding to [%s] from [%s]",
```

Comment:
nit: can you add what causes this override to the log so that it's clearer when reading the log?

Reply:
Also I feel the debug level should be warning/info? Not quite sure 🤔

Reply:
updated & upgraded